### PR TITLE
BartWorks support for apiculture requirements

### PR DESCRIPTION
**Pull request by ElNounch**
Fix Salt bee from breeding Clay + Aluminium.
Should fix GTNewHorizons/NewHorizons#5096

### CODE REVIEW

**Comment by Dream-Master** [comment]
@ElNounch what is your Name in GTNH Discord? Want to hear from you what this pull fixed? Pm me. Thanks

File: src/main/java/forestry/core/utils/BlockUtil.java [hunk 1]
```diff
@@ -228,4 +234,70 @@ private static boolean isVecInsideXZBounds(Vec3 vec, float minX, float minZ, flo
...
+
+	private static boolean searchedForeignClasses = false;
+	private static void grabForeignClasses() {
+		if (searchedForeignClasses) {
+			return;
+		}
+		try {
+			cl_BW_MetaGenerated_Blocks = Class.forName("com.github.bartimaeusnek.bartworks.system.material.BW_MetaGenerated_Blocks");
+			Class cl_BW_MetaGenerated_Block_TE = Class.forName("com.github.bartimaeusnek.bartworks.system.material.BW_MetaGenerated_Block_TE");
+			for (Method mt : cl_BW_MetaGenerated_Block_TE.getDeclaredMethods()) {
```
Comment (draknyte1) [inline src/main/java/forestry/core/utils/BlockUtil.java]: No point iterating all methods if you know what method you want.. just grab it directly 🤷🏻‍♀️

Comment (ElNounch) [reply src/main/java/forestry/core/utils/BlockUtil.java]: getMethod only works on public ones.

Comment (draknyte1) [reply src/main/java/forestry/core/utils/BlockUtil.java]: So use getDeclaredMethod... don’t need to iterate them all 😂😂

**Comment by wlhlm** [comment]
This looks to be problematic: GTNewHorizons/NewHorizons#5816

**Comment by bartimaeusnek** [comment]
nah just a stack.equals issue instead of a util compareison... 

fixed with
https://github.com/GTNewHorizons/ForestryMC/commit/5db5b712ceb842acf7f6b575092a1bfd1c7c2a34

**Pull request merged**